src/status_im/contexts/wallet/add_account/add_address/view.cljs

```diff
@@ -7,28 +7,38 @@
     [reagent.core :as reagent]
     [status-im.common.floating-button-page.view :as floating-button-page]
     [status-im.constants :as constants]
-    [status-im.contexts.wallet.add-account.add-address-to-watch.style :as style]
+    [status-im.contexts.wallet.add-account.add-address.style :as style]
     [status-im.contexts.wallet.common.validation :as validation]
     [status-im.subs.wallet.add-account.address-to-watch]
     [utils.debounce :as debounce]
     [utils.i18n :as i18n]
     [utils.re-frame :as rf]))
 
+(def ^:private adding-addresses-purposes
+  {:watch {:title       :t/add-address-to-watch
+           :description :t/enter-eth
+           :input-title :t/eth-or-ens}
+   :save  {:title       :t/add-address
+           :description :t/save-address-description
+           :input-title :t/address-or-end-name}})
+
 (defn- validate-address
-  [known-addresses user-input]
+  [known-addresses user-input purpose]
```

Comment:
Imo Better to pass the text (i18n label) than this generic purpose which is rather specific here